src/test/java/harness/utils/SamGroups.java

```diff
@@ -0,0 +1,46 @@
+package harness.utils;
+
+import harness.TestCommand;
+import harness.TestUsers;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+/**
+ * Utility class for tracking groups created by a test class, and then trying to delete them all at
+ * the end.
+ */
+public class SamGroups {
+  // keep a map of groups created by tests, so we can try to clean them up
+  // group name -> test user that created it
+  private Map<String, TestUsers> trackedGroups = new HashMap<>();
+
+  private static final Random RANDOM = new Random();
+
+  /** Try to delete each group that was tracked here. */
+  public void deleteAllTrackedGroups() throws IOException {
+    for (Map.Entry<String, TestUsers> groupCreated : trackedGroups.entrySet()) {
+      groupCreated.getValue().login();
+      TestCommand.Result cmd =
+          TestCommand.runCommand("groups", "delete", "--name=" + groupCreated.getKey());
+      if (cmd.exitCode == 0) {
+        System.out.println("group was not cleaned up by test: " + groupCreated.getKey());
```

Comment:
is this going to log if the group was previously deleted? maybe worth a comment either way.

Reply:
It will only log here if the group was not previously deleted. Added a comment.